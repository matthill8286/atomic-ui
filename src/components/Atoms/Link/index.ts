export * from './Link'
export * from './Link.interface'
