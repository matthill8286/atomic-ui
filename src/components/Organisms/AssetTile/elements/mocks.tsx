import { newAssetMocks, newStaticAssetMocks } from './mockAsset'
import { AssetTileLayout } from '@/components/Organisms/AssetTile/AssetTile.interface'

export const layout: AssetTileLayout[] = [
  AssetTileLayout.gridItem,
  AssetTileLayout.listItem,
  AssetTileLayout.compact,
  AssetTileLayout.auto,
]

export const mockAssets = {
  asset1: newAssetMocks[0],
  asset2: newAssetMocks[1],
  asset3: newAssetMocks[2],
  asset4: newAssetMocks[4], // screenshotOverride empty test, uses image as fallback
}

export const staticMockAssets = {
  asset1: newStaticAssetMocks[0],
  asset2: newStaticAssetMocks[1],
  asset3: newStaticAssetMocks[2],
  asset4: newStaticAssetMocks[3],
}

export const gridSelection = {
  '1 Item': 1,
  '2 Items': 2,
  '3 Items': 3,
}

export const assetNames = ['asset1', 'asset2', 'asset3']

export const assetOptions = {
  '(Automatic)': 'asset1',
  asset1: 'asset1',
  asset2: 'asset2',
  asset3: 'asset3',
}

export const featureOptions = {
  showSkills: 'showSkills',
  showBookmark: 'showBookmark',
  showLock: 'showLock',
  showProvider: 'showProvider',
  showAssetMeta: 'showAssetMeta',
  showFeatured: 'showFeatured',
}

export const baseProps = {}

export const presetProps = ['None', 'Reduced']
