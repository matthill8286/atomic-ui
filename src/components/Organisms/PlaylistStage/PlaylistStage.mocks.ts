import { BreadcrumbPath } from '@/components/Molecules/Breadcrumb/Breadcrumb.interface'
import { VerticalPaddingMap } from '@/types/theme'
import { PlaylistStageProps, PlaylistProps } from './PlaylistStage.interface'
import { getTransformedImageVersion } from '@/utils/useGraphCmsImages'

export const backlink: BreadcrumbPath = {
  name: "Let's Go home",
  link: '/home',
}

export const breadcrumbPath: BreadcrumbPath[] = [
  {
    name: 'Playlist',
    link: '/',
  },
  {
    name: 'Collections',
    link: '/',
  },
]

export const height = {
  mobile: 200,
  tablet: 260,
  desktop: 300,
}

export const image = {
  mobile: getTransformedImageVersion('1FfvDDFERwWh35wkpNky'),
  tablet: getTransformedImageVersion('zdkfrHcOQQmNekeIxjve'),
}

export const paddingBottom: VerticalPaddingMap = {
  mobile: 'xs',
  tablet: 'xs',
}

export const paddingTop: VerticalPaddingMap = {
  tablet: 'xs',
  desktop: 'xs',
}

export const PlaylistMock: PlaylistProps = {
  pid: '628ce219-0158-4eb0-b864-61c59eff0f70',
  name: 'Conflict Management',
  image: getTransformedImageVersion('FtMfGctDTKGIYVzDm5dL'),
  strategy: '?',
  totalAssetDuration: '1 day 15 hours 31 minutes',
  completionPercentage: '75% completed',
  numberOfAssets: '12 assets',
  description: 'This is the Conflict Management playlist',
}

export const props: PlaylistStageProps = {
  playlist: PlaylistMock,
  breadcrumbPath: undefined,
  colors: 'primary',
  headingColor: 'grey6',
  height,
  paddingBottom,
  paddingTop,
  withImage: true,
  shape: 'rounded-small',
}
