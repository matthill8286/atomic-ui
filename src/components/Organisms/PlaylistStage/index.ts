export * from './PlaylistStage'
export * from './PlaylistStage.interface'
export * from './PlaylistStage.mocks'
