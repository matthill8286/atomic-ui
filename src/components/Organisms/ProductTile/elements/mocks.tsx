import { newProductMocks, newStaticProductMocks } from './mockProduct'
import { ProductTileLayout } from '@/components/Organisms/ProductTile/ProductTile.interface'

export const layout: ProductTileLayout[] = [
  ProductTileLayout.gridItem,
  ProductTileLayout.listItem,
  ProductTileLayout.compact,
  ProductTileLayout.auto,
]

export const mockProducts = {
  product1: newProductMocks[0],
  product2: newProductMocks[1],
  product3: newProductMocks[2],
  product4: newProductMocks[4], // screenshotOverride empty test, uses image as fallback
}

export const staticMockProducts = {
  product1: newStaticProductMocks[0],
  product2: newStaticProductMocks[1],
  product3: newStaticProductMocks[2],
  product4: newStaticProductMocks[3],
}

export const gridSelection = {
  '1 Item': 1,
  '2 Items': 2,
  '3 Items': 3,
}

export const productNames = ['product1', 'product2', 'product3']

export const productOptions = {
  '(Automatic)': 'product1',
  product1: 'product1',
  product2: 'product2',
  product3: 'product3',
}

export const featureOptions = {
  showSkills: 'showSkills',
  showBookmark: 'showBookmark',
  showLock: 'showLock',
  showProvider: 'showProvider',
  showProductMeta: 'showProductMeta',
  showFeatured: 'showFeatured',
}

export const baseProps = {}

export const presetProps = ['None', 'Reduced']
