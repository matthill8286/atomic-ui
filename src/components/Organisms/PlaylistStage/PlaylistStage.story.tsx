import { text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { PlaylistStage } from './PlaylistStage'
import { PlaylistStageProps } from './PlaylistStage.interface'
import { backlink, props, PlaylistMock } from './PlaylistStage.mocks'

const knobs = (): Partial<PlaylistStageProps> => {
  return {
    playlist: {
      ...PlaylistMock,
      name: text('Article Title', 'Our Playlist'),
      description: text('Subtext', 'This is a description'),
    },
  }
}

storiesOf('Design System/Organisms/Playlist Stage', module)
  .add('Default', () => {
    return <PlaylistStage {...props} {...knobs()} />
  })
  .add('Without Image', () => {
    return <PlaylistStage {...props} {...knobs()} withImage={false} />
  })
  .add('Without Breadcrumbs', () => {
    return <PlaylistStage {...props} {...knobs()} breadcrumbPath={undefined} />
  })
  .add('With Backlink', () => {
    return <PlaylistStage {...props} {...knobs()} breadcrumbPath={undefined} backlink={backlink} />
  })
  .add('Different Headline Colour', () => {
    return <PlaylistStage {...props} {...knobs()} headingColor="black" />
  })
