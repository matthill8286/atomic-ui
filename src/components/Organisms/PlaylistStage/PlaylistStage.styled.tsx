import { Heading } from '@/components/Atoms/Typography'
import { FlexBox } from '@/components/Helper/FlexBox'
import { Grid } from '@/components/Helper/Grid'
import { css, media, styled } from '@/styles'
import { StyledHeadingArticleProps, StyledPlaylistStageTitleProps } from './PlaylistStage.interface'
import { ProductTileMeta } from '@/components/Organisms/ProductTile/elements'

const articleStageBackgroundCommonStyles = css`
  content: ' ';
  display: none;
  position: absolute;
  width: 100%;
  top: 0;
  z-index: -1;

  ${media.sm} {
    display: block;
  }
`

export const StyledProgressWrapper = styled.div<{ isSmall?: boolean }>`
  position: absolute;
  bottom: ${({ isSmall }) =>
    isSmall
      ? '-4px'
      : '-8px'}; // this is minus the height of the progress so it truly is at the bottom
  left: 0;
  right: 0;
`

export const StyledPlaylistStageTitle = styled.div<StyledPlaylistStageTitleProps>`
  ${({ theme, withImage, isRoundBorder, isSkewedBorder }) => {
    const { color, dimension, spacing } = theme
    const { borderRadius1 } = dimension
    const { selected, white } = color
    const { md } = spacing.base

    return css`
      background-color: ${withImage ? white : 'transparent'};
      box-sizing: border-box;
      color: ${selected};
      padding: ${withImage ? md : 0};
      position: relative;
      max-width: 780px;
      margin: 0 auto;
      z-index: 1;
      border-radius: 0 ${borderRadius1} ${borderRadius1} 0;
      box-shadow: ${withImage ? '8px 0 8px 8px rgba(0, 0, 0, 0.07);' : 'none'};

      ${media.sm} {
        width: calc(100% - 71px);
        border-top-right-radius: ${160 / 25}px;
        border-bottom-right-radius: ${160 / 5}px;
        box-shadow: ${withImage ? '8px 0 8px 8px rgba(0, 0, 0, 0.07);' : 'none'};
      }

      ${media.md} {
        border-radius: ${borderRadius1};
        box-shadow: ${withImage ? '8px 0 8px 8px rgba(0, 0, 0, 0.07);' : 'none'};

        h1 {
          height: inherit;
        }
      }

      ${isRoundBorder &&
        css`
          &:after {
            ${articleStageBackgroundCommonStyles}
            transform: skew(-19.7deg);
            height: 100%;
            right: -50px;
            border-top-right-radius: ${160 / 25}px;
            border-bottom-right-radius: ${160 / 5}px;
            box-shadow: ${withImage ? '8px 0 8px 0 rgba(0, 0, 0, 0.07);' : 'none'};
            background: ${white};
          }
        `}

      ${isSkewedBorder &&
        css`
          &:before {
            ${articleStageBackgroundCommonStyles}
            transform: skew(25deg);
            height: 50%;
            right: -50px;
            background: ${withImage ? white : 'transparent'};
          }

          &:after {
            ${articleStageBackgroundCommonStyles}
            transform: skew(-25deg);
            height: 50%;
            top: 50%;
            right: -50px;
            box-shadow: ${withImage ? '8px 8px 8px 0 rgba(0, 0, 0, 0.07)' : 'none'};
            background: ${withImage ? white : 'transparent'};
          }
        `}
    `
  }}
`

export const StyledPlaylistStageGrid = styled(Grid)`
  position: absolute;
  left: 0;
  right: 0;
  bottom: -46px;
`

export const StyledPlaylistStage = styled.div`
  ${({ theme }) => {
    return css`
      margin-bottom: ${theme.spacing.base.xxxl};
      position: relative;
    `
  }}
`

export const StyledHeadingArticle = styled(Heading)<StyledHeadingArticleProps>`
  ${({ theme, breadcrumbPath }) => {
    return css`
      margin: 0;
      font-size: ${theme.font.size.xl};

      ${media.md} {
        margin: 0 0 ${breadcrumbPath ? theme.spacing.base.sm : 0} 0;
        font-size: ${theme.font.size.lg};
        line-height: ${theme.spacing.base.lg};
      }

      ${media.mobile} {
        margin: 0 0 ${breadcrumbPath ? theme.spacing.base.xs : 0} 0;
        font-size: ${theme.font.size.sm};
        line-height: ${theme.spacing.base.md};
      }
    `
  }}
`

export const StyledBreadcrumbWrapper = styled.div`
  display: none;

  ${media.md} {
    display: block;
  }
`

export const StyledBacklinkWrapper = styled(FlexBox)`
  display: flex;
  margin: 0 0 8px;

  & > *:not(:first-child) {
    margin: 0 4px;
  }
`

export const StyledProductTileMetaWrapper = styled(FlexBox)`
  margin: 0 auto;
  ${media.maxSm} {
    margin: 0 auto;
  }
`

export const StyledContentWrapper = styled(FlexBox)`
  max-width: 80%;
  margin: 0 auto;
  padding-left: ${({ theme }) => theme.spacing.base.lg};
  padding-right: ${({ theme }) => theme.spacing.base.lg};

  ${media.maxMd} {
    margin: ${({ theme }) => theme.spacing.base.xs} auto;
    padding: 0;
    width: 100%;
  }
`
