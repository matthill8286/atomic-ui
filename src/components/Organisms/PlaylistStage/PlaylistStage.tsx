import React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { Link } from '@/components/Atoms/Link'
import { Section } from '@/components/Atoms/Section'
import { CopyText } from '@/components/Atoms/Typography'
import { Breadcrumb } from '@/components/Molecules/Breadcrumb'
import { IconArrow } from '@excelwithbusiness/webmobile-svg-library'
import { ImageMap } from '@/types'
import { isFilteredTheme } from '@/utils/helper'
import { PlaylistStageProps } from './PlaylistStage.interface'
import {
  StyledAssetTileMetaWrapper,
  StyledBacklinkWrapper,
  StyledBreadcrumbWrapper,
  StyledContentWrapper,
  StyledHeadingArticle,
  StyledPlaylistStage,
  StyledPlaylistStageGrid,
  StyledPlaylistStageTitle,
  StyledProgressWrapper,
} from './PlaylistStage.styled'
import { AssetTileMeta } from '@/components/Organisms/AssetTile/elements'
import { getMetaItemList } from '@/components/Organisms/AssetTile/helpers/icons'
import { ProgressBar } from '@/components/Atoms/ProgressBar'

export const PlaylistStage: React.FC<PlaylistStageProps> = ({
  playlist,
  backlink,
  breadcrumbPath,
  colors,
  headingColor = 'selected',
  height,
  paddingBottom,
  paddingTop,
  withImage = true,
  withMetaItems = true,
  homeLink,
  shape,
  ...rest
}) => {
  const {
    name,
    image,
    strategy,
    totalAssetDuration,
    completionPercentage,
    numberOfAssets,
    description,
  } = playlist

  const metaItems = getMetaItemList({
    duration: totalAssetDuration,
    type: undefined,
    completedStatus: completionPercentage,
    numberOfAssets,
  })

  const imageMap: ImageMap = {
    mobile: image as string,
    tablet: image,
    desktop: image,
  }

  const PlaylistStageContent = React.useCallback(() => {
    return (
      <StyledPlaylistStageGrid>
        <StyledPlaylistStageTitle
          withImage={withImage}
          isSkewedBorder={false}
          isRoundBorder={!isFilteredTheme()}>
          <StyledContentWrapper
            justifyContent="center"
            flexDirection="column"
            alignItems="space-between">
            {backlink && (
              <StyledBacklinkWrapper>
                <Icon color="grey6" rotate={90} width="sm">
                  <IconArrow />
                </Icon>
                <Link href={backlink.link} color="grey6" fontSize="sm" inline>
                  {backlink.name}
                </Link>
              </StyledBacklinkWrapper>
            )}
            <StyledHeadingArticle
              breadcrumbPath={breadcrumbPath}
              color={headingColor}
              fontFamily="default"
              weight="bold"
              fontSize="xl"
              lineHeight="xxl"
              textAlign="center"
              scale="level-2"
              tag="h2">
              {name}
            </StyledHeadingArticle>
            <StyledBreadcrumbWrapper>
              {breadcrumbPath && <Breadcrumb homeLink={homeLink} paths={breadcrumbPath} />}
            </StyledBreadcrumbWrapper>
            {description && (
              <CopyText as="div" margin="sm 0" fontSize="sm" textAlign="center" limitLines={2}>
                {description}
              </CopyText>
            )}
          </StyledContentWrapper>
          {withMetaItems && (
            <>
              <StyledAssetTileMetaWrapper justifyContent="center" flexDirection="row">
                <AssetTileMeta inPlaylist justify="center" loading={false} metaItems={metaItems} />
              </StyledAssetTileMetaWrapper>
              <StyledProgressWrapper isSmall>
                <ProgressBar value={parseInt(completionPercentage, 0)} small />
              </StyledProgressWrapper>
            </>
          )}
        </StyledPlaylistStageTitle>
      </StyledPlaylistStageGrid>
    )
  }, [name, breadcrumbPath, headingColor, description])

  return (
    <StyledPlaylistStage {...rest}>
      {withImage ? (
        <Section
          color={colors}
          height={height}
          image={imageMap}
          paddingBottom={paddingBottom}
          paddingTop={paddingTop}
          shape={shape}>
          <PlaylistStageContent />
        </Section>
      ) : (
        <PlaylistStageContent />
      )}
    </StyledPlaylistStage>
  )
}
