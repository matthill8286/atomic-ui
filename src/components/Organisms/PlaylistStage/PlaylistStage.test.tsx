import React from 'react'
import { Link } from '@/components/Atoms/Link'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { PlaylistStage } from './PlaylistStage'
import { backlink, props } from './PlaylistStage.mocks'
import { StyledBacklinkWrapper } from './PlaylistStage.styled'

describe('<PlaylistStage />', () => {
  test('Renders without error', () => {
    const tree = renderWithTheme(<PlaylistStage {...props} />)
    expect(tree).toMatchSnapshot()
  })

  describe('Backlink', () => {
    test('should be hidden by default', () => {
      const wrapper = mountWithTheme(<PlaylistStage {...props} />)
      expect(wrapper.find(StyledBacklinkWrapper)).toHaveLength(0)
    })

    test('should be visible when backlink prop provided', () => {
      const wrapper = mountWithTheme(<PlaylistStage {...props} backlink={backlink} />)

      expect(wrapper.find(StyledBacklinkWrapper)).toHaveLength(1)
      const link = wrapper.find(StyledBacklinkWrapper).find(Link)
      expect(link?.prop('href')).toEqual(backlink.link)
      expect(link?.contains(backlink.name)).toEqual(true)
    })
  })
})
