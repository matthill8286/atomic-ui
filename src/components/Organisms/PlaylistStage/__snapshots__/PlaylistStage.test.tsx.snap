// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaylistStage /> Renders without error 1`] = `
.c7 {
  margin: 0;
  color: #333333;
  font-size: 26px;
  line-height: 44px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  text-align: center;
}

.c10 {
  margin: 16px 0;
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
  white-space: normal;
  overflow: hidden;
  max-height: calc(24px * 2);
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.c1 {
  width: 100%;
  height: 200px;
  left: 0;
  right: 0;
  background: #00C8DC no-repeat center center;
  background-image: url(getTransformedImageVersion(FtMfGctDTKGIYVzDm5dL));
  background-size: cover;
  padding-bottom: 8px;
  border-radius: 8px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
}

.c2 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

.c4 {
  background-color: #FFFFFF;
  box-sizing: border-box;
  color: #4bcade;
  padding: 24px;
  position: relative;
  max-width: 780px;
  margin: 0 auto;
  z-index: 1;
  border-radius: 0 4px 4px 0;
  box-shadow: 8px 0 8px 8px rgba(0,0,0,0.07);
}

.c3 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: -46px;
}

.c0 {
  margin-bottom: 56px;
  position: relative;
}

.c8 {
  margin: 0;
  font-size: 26px;
}

.c9 {
  display: none;
}

.c6 {
  max-width: 80%;
  margin: 0 auto;
  padding-left: 32px;
  padding-right: 32px;
}

@media (min-width:752px) {
  .c1 {
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media (min-width:1008px) {
  .c1 {
    padding-top: 8px;
  }
}

@media (min-width:752px) {
  .c1 {
    background-image: url(getTransformedImageVersion(FtMfGctDTKGIYVzDm5dL));
    height: 260px;
  }
}

@media (min-width:1008px) {
  .c1 {
    background-image: url(getTransformedImageVersion(FtMfGctDTKGIYVzDm5dL));
    height: 300px;
  }
}

@media (min-width:512px) {
  .c2 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c2 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c2 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c2 {
    padding: 0 64px;
  }
}

@media (min-width:512px) {
  .c4 {
    width: calc(100% - 71px);
    border-top-right-radius: 6.4px;
    border-bottom-right-radius: 32px;
    box-shadow: 8px 0 8px 8px rgba(0,0,0,0.07);
  }
}

@media (min-width:752px) {
  .c4 {
    border-radius: 4px;
    box-shadow: 8px 0 8px 8px rgba(0,0,0,0.07);
  }

  .c4 h1 {
    height: inherit;
  }
}

@media (min-width:752px) {
  .c8 {
    margin: 0 0 0 0;
    font-size: 22px;
    line-height: 32px;
  }
}

@media (max-width:512px) {
  .c8 {
    margin: 0 0 0 0;
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:752px) {
  .c9 {
    display: block;
  }
}

@media (max-width:752px) {
  .c6 {
    margin: 8px auto;
    padding: 0;
    width: 100%;
  }
}

<div
  className="c0"
>
  <section
    className="c1"
    color="primary"
    height={
      Object {
        "desktop": 300,
        "mobile": 200,
        "tablet": 260,
      }
    }
    shape="rounded-big"
  >
    <div
      className="c2 c3"
    >
      <div
        className="c4"
      >
        <div
          className="c5 c6"
          display="flex"
        >
          <h2
            className="c7 c8"
            color="grey6"
            fontFamily="default"
            fontSize="xl"
            spacing="base"
          >
            Conflict Management
          </h2>
          <div
            className="c9"
          />
          <div
            className="c10"
            color="grey5"
            fontSize="sm"
          >
            This is the Conflict Management playlist
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
